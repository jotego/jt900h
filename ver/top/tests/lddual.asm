    ; LD data read from memory into memory
    main section code
    org 0
    ld a,0xbf    ; common header

    ld a,0

    ldw (data),0x1234
    ld xix,data
    ld de,(xix)
    cp de,0x1234
    jp ne,bad_end

    incf
    ld (data),0x55
    ld d,(xix)
    cp d,0x55
    jp ne,bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
    align 2
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    dw 0,0,0,0,0,0,0,0,0,0
    align 4
stack:
    end