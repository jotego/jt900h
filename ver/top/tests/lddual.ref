00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00001234 - 00005500 - 00000000 - 00000000
00000000 - 0000babe - 00000000 - 00000000
-- index registers --
00000044 - 00000000 - 00000000 - 00000100
SR = f942
