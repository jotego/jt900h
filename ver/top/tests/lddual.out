00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00001234 - 00005500 - 00000000 - 00000000
00000000 - 0000babe - 00000000 - 00000000
-- index registers --
00000040 - 00000000 - 00000000 - 00000000
SR = f942
